Sources/MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -130,30 +146,62 @@ open class SimulatedLocationManager: NavigationLocationManager {
     
     private var routeProgress: RouteProgress?
     
+    private var _nextDate: Date? = nil
+    private func getNextDate() -> Date {
+        if _nextDate == nil || _nextDate! < Date() {
+            _nextDate = Date()
+        } else {
+            _nextDate?.addTimeInterval(1)
+        }
+        return _nextDate!
+    }
+    
+    private var slicedIndex: Int? = nil
+    
     internal func tick() {
-        guard let polyline = routeShape, let newCoordinate = polyline.coordinateFromStart(distance: currentDistance) else {
+        guard let polyline = routeShape,
+              let indexedNewCoordinate = polyline.indexedCoordinateFromStart(distance: currentSpeed) else {
```

Comment:
Given that we emulate each tick as 1 second, each tick distance is the current speed. When speeding up the simulation we are not incrementing distance or speed travelled with each tick. Instead we produce ticks more often, maintaining it's timestamp to monotonically increment.

Reply:
I see, but it really lloks strange when you use speed instead of distance even if numeric values are the same. My concern is not about the calculation is wrong, but that the variable name used in calculation is confusing.

Reply:
added `tickDistance` for clarity